controllers/workspace/status.go

```diff
@@ -47,6 +47,31 @@ func initCurrentStatus() currentStatus {
 	}
 }
 
+// initStoppingStatus initialize the status for stopping workspace
+// it may be STOPPING or FAILED in case it's stopped automatically due workspace startup failure
+// in the second case the corresponding condition is propagated
+func initStoppingStatus(dwStatus dw.DevWorkspaceStatus) currentStatus {
```

Comment:
This seems like a confusing abstraction for a very specific we need to do. Perhaps a better name would help?

In general, if a function is only going to be called in one place, I don't see much benefit in breaking it out like this.

Reply:
Well, initStartingStatus is called also from one place.
It's also called finalize.go but it has a different use-case because it gets STARTING status and then update phase to Error, so starting does not make sense.

I tried to play with it a bit and @amisevsk WDYT if we:
1. Introduce method `New` that will init conditions.
2. Separate actual workspace stopping logic from managing status, to avoid `changeToStoppedIfNeeded` func as well as doing the same checks in two places

Like the following https://github.com/sleshchenko/devworkspace-operator/commit/caa5a68ea17bd1261b6f67723067a1c6c3d879e0

stop... doStop... may be not the best naming
doStop came to me from Java but in this context `scaleDeploymentDown` may be better at least until we don't do anything apart from scaling deployment to zero on stopping workspaces.

Reply:
That makes sense to me -- also I would rework `initCurrentStatus` to just remove it -- all we really need to do is make sure the map isn't `nil`. It's safer to just do that in the functions that write to it, e.g. 
```go
func (c *workspaceConditions) setConditionTrue(conditionType dw.DevWorkspaceConditionType, msg string) {
	if c.conditions == nil {
		c.conditions = map[dw.DevWorkspaceConditionType]dw.DevWorkspaceCondition{}
	}
	c.conditions[conditionType] = dw.DevWorkspaceCondition{
		Status:  corev1.ConditionTrue,
		Message: msg,
	}
}
```

> Well, initStartingStatus is called also from one place.

It used to be called `initCurrentStatus` and was meant to be used in a different way :)